RULES/libsupcxx/xlib/xlib.go

```diff
@@ -0,0 +1,76 @@
+package xlib
+
+import (
+	"fmt"
+	"os"
+	"path"
+
+	"dbt-rules/RULES/cc"
+	"dbt-rules/RULES/core"
+
+	"libsupcxx/RULES/target"
+	"libsupcxx/libsupcxx/include"
+)
+
+// This package is only meant to be used within libsupcxx, outside code should use
+// libsupcxx/RULES/libsupcxx. See there for info about XLibrary.
+type XLibrary struct {
+	Out           core.OutPath
+	Srcs          []core.Path
+	Includes      []core.Path
+	CompilerFlags []string
+	XDeps         []XLibrary
+	Shared        bool
+	AlwaysLink    bool
+	Targets       []target.Target
+}
+
+func ContainsTarget(ts []target.Target, t target.Target) bool {
+	for _, el := range ts {
+		if el.Name == t.Name {
+			return true
+		}
+	}
+	return false
+}
+
+func OutPathForTarget(base core.OutPath, t target.Target) core.OutPath {
+	return base.WithPrefix("/" + path.Base(base.Absolute()) + "-" + t.Name + "/")
```

Comment:
why are you using `base.Absolute` here? Shouln't relative be enough?

Reply:
yes, it is